### ISSUE TITLE
Accessibility role on MenuItem should be on `a` inside the `li`

### ISSUE THREAD

**Issue by cdeutsch** [ant-design/ant-design#10095]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.4.1

### Environment
Chrome Version 65.0.3325.181 (Official Build) (64-bit) on OSX

### Reproduction link
[https://ant.design/components/menu/](https://ant.design/components/menu/)

### Steps to reproduce
Use Chrome Developer tools to create an Accessibility Audit.

### What is expected?
Should pass audit.

### What is actually happening?
`[aria-*] attributes do not match their roles.`

According to this the `role="menuitem"` should be on an `a` element inside the `li` instead of on the `li` itself
https://www.w3.org/TR/2017/NOTE-wai-aria-practices-1.1-20171214/examples/menu-button/menu-button-links.html

![image](https://user-images.githubusercontent.com/404765/38830749-e7b32924-4182-11e8-92c8-b48fc24d023d.png)

**Comment by picodoth** [comment]
looking

**Comment by picodoth** [comment]
For the examples, we actually don't have any `a` tag in MenuItem, but looking at https://www.w3.org/TR/2017/NOTE-wai-aria-practices-1.1-20171214/examples/menu-button/menu-button-links.html, I think it still makes sense. We might consider conditionally apply role="menuitem" denpending on the children type.

The actual violation is aria-selected, according to https://www.levelaccess.com/how-not-to-misuse-aria-states-properties-and-roles/
```
Only use aria-selected with these roles: option, tab, menuitemradio, treeitem, gridcell, row, rowheader and columnheader. Currently the ARIA spec indicates aria-selected can be used with radio buttons, but this is not a correct use of the property. When authors want to indicate that a radio button is checked they should use the aria-checked state –  in general, the aria-selected state should not be used with radio buttons.
```
I removed aria-selected and the corresponding complains is gone. But I need to gather more info on why we added that in the first place.

We also have some issue with aria-own and aria-activedescendant.

**Comment by picodoth** [comment]
@cdeutsch I have two ongoing pr to address the issue. But please note that in the case of nav items like:
```html
<li>
  <a role="menuitem">link</a>
</li>
```
We leave it to the consumer for giving `role="menuitem"`:
```html
<MenuItem role={null}>
  <a role="menuitem">link</a>
</MenuItem>
```
The reason is mainly  to keep convenient default aria-* value for MenuItem and to avoid introducing breaking changes.

**Comment by cdeutsch** [comment]
That should work! Thanks!!

**Comment by yesmeck** [comment]
@paranoidjk Has this problem been solved in rc-menu?

**Comment by cdeutsch** [comment]
@yesmeck yes, it worked for me 👍
